lib/ecto/changeset.ex

```diff
@@ -506,10 +580,16 @@ defmodule Ecto.Changeset do
   """
   @spec put_change(t, atom, term) :: t
   def put_change(%Changeset{} = changeset, key, value) do
```

Comment:
Let's add a clause that checks here if `types: nil` and, if so, we raise a nice error message.